api/jobs/handlers.py

```diff
@@ -74,7 +74,20 @@ def post(self, _id):
         if _id != doc.get('name', ''):
             self.abort(400, 'Name key must be present and match URL')
 
-        upsert_gear(self.request.json)
+        try:
+            upsert_gear(self.request.json)
+        except ValidationError as err:
+            key = None
+            if len(err.relative_path) > 0:
+                key = err.relative_path[0]
+
+            self.response.set_status(400)
```

Comment:
You use a 422 for a schema validation error in jobs/add - we should keep it consistent.